### PR TITLE
Take all dependencies from Hackage or CHaP

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

**Comment by newhoggy** [comment]
Needs rebase, and seems like there is a Conway related configuration error that's failing the tests.

**Comment by angerman** [comment]
This PR needs #4871 to be merged first to support CHaP revisions :-/

File: cardano-api/src/Cardano/Api/LedgerState.hs [hunk 1]
```diff
@@ -66,7 +66,8 @@ import           Data.Aeson.Types (Parser)
 import           Data.Bifunctor
 import           Data.ByteArray (ByteArrayAccess)
 import qualified Data.ByteArray
-import           Data.ByteString as BS
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/LedgerState.hs:69]: :face_with_spiral_eyes:

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 2]
```diff
@@ -502,14 +515,37 @@ instance IsShelleyBasedEra era => FromJSON (TxOut CtxTx era) where
...
+                (_,_) -> fail "Should not be possible to create a tx output with either an inline datum hash or an inline datum"
+
+            mReferenceScript <- o .:? "referenceScript"
+
+            reconcileConway alonzoTxOutInConway mInlineDatum mReferenceScript
          where
-           reconcile
+           reconcileBabbage
              :: TxOut CtxTx BabbageEra -- ^ Alonzo era datum in Babbage era
              -> TxOutDatum CtxTx BabbageEra -- ^ Babbagae inline datum
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/TxBody.hs:545]: Can you fix this typo while you're here?  `s/Babbagae/Babbage`

Also all other occurrences in this file.

Comment (erikd) [reply cardano-api/src/Cardano/Api/TxBody.hs:545]: Done!

**Review by newhoggy — APPROVED** [review]

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 3]
```diff
@@ -502,14 +515,37 @@ instance IsShelleyBasedEra era => FromJSON (TxOut CtxTx era) where
...
+                      else return $ TxOutDatumInline ReferenceTxInsScriptsInlineDatumsInConwayEra sData
+                (Nothing, Nothing) -> return TxOutDatumNone
+                (_,_) -> fail "Should not be possible to create a tx output with either an inline datum hash or an inline datum"
+
+            mReferenceScript <- o .:? "referenceScript"
+
+            reconcileConway alonzoTxOutInConway mInlineDatum mReferenceScript
          where
-           reconcile
+           reconcileBabbage
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/TxBody.hs:543]: :+1:

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs [hunk 4]
```diff
@@ -215,6 +228,15 @@ mkSomeConsensusProtocolCardano NodeByronProtocolConfiguration {
           Praos.babbageMaxTxCapacityOverrides =
             TxLimits.mkOverrides TxLimits.noOverridesMeasure
         }
+        Praos.ProtocolParamsConway {
+          -- This is /not/ the Babbage protocol version. It is the protocol
+          -- version that this node will declare that it understands, when it
+          -- is in the Babbage era. Since Babbage is currently the last known
+          -- protocol version then this is also the Babbage protocol version.
+          Praos.conwayProtVer = ProtVer 9 0,
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Protocol/Cardano.hs:236]: Shouldn't we be hiding this behind the `TestEnableDevelopmentHardForkEras` flag: https://github.com/input-output-hk/cardano-node/commit/e4240315820ce55ec70e718610ed63e7e380fc84#diff-8c649864fe4b96d142cfd4c2bb9e32a9510fe737c6ad4e13e04b585ee15a6b74R210

**Pull request merged**